clang/lib/Sema/SemaExpr.cpp

```diff
@@ -13329,6 +13359,9 @@ inline QualType Sema::CheckLogicalOperands(ExprResult &LHS, ExprResult &RHS,
   // The following is safe because we only use this method for
   // non-overloadable operands.
 
+  if (IsAMDGPUPredicateBI(LHS.get()) && IsAMDGPUPredicateBI(RHS.get()))
```

Comment:
If we want to diagnose this not in the 'right' place we should, though that is going ot be absolutely fraught with problems.  But making it return `void` is not the way to do this.